lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2362,6 +2362,200 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
+    associations.push_back(ReassociationIndices{2, 3});
+    SmallVector<int64_t> gridShapeCollapsed{gridShape[0], gridShape[1],
+                                            gridShape[2]};
+    gridShapeExtracted.back() = 1;
+    auto gridTypeCollapsed =
+        RankedTensorType::get(gridShapeCollapsed, gridElementType);
+    auto gridCollapsed_0 = rewriter.create<tensor::CollapseShapeOp>(
+        loc, gridTypeCollapsed, grid_0, associations);
+    auto gridCollapsed_1 = rewriter.create<tensor::CollapseShapeOp>(
+        loc, gridTypeCollapsed, grid_1, associations);
+    AffineExpr d0 = getAffineDimExpr(0, op->getContext());
+    AffineExpr d2 = getAffineDimExpr(2, op->getContext());
+    AffineExpr d3 = getAffineDimExpr(3, op->getContext());
+    SmallVector<AffineMap> gridMaps{
+        rewriter.getMultiDimIdentityMap(gridRank),
+        AffineMap::get(4, 0, {d0, d2, d3}, op->getContext()),
+        AffineMap::get(4, 0, {d0, d2, d3}, op->getContext())};
+    SmallVector<utils::IteratorType> gridIterators(
+        gridRank, utils::IteratorType::parallel);
+    SmallVector<int64_t> resultShape = makeShapeTorchCompatible(inputShape);
+    resultShape[2] = gridShape[1];
+    resultShape[3] = gridShape[2];
+    Value resultFinal =
+        rewriter.create<tensor::EmptyOp>(loc, resultShape, floatType);
+    auto resultFinalType = resultFinal.getType().cast<RankedTensorType>();
+    Value sGrid =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, resultFinalType,
+                ValueRange{gridCollapsed_0, gridCollapsed_1}, resultFinal,
+                gridMaps, gridIterators,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value gr_0 = args[0];
+                  Value gr_1 = args[1];
+                  Value oneFloat = b.create<arith::ConstantOp>(
+                      loc, b.getFloatAttr(gridElementType, 1.0));
+                  Value gplus_0 = b.create<arith::AddFOp>(loc, gr_0, oneFloat);
+                  Value gplus_1 = b.create<arith::AddFOp>(loc, gr_1, oneFloat);
+                  Value result_0 =
+                      b.create<arith::MulFOp>(loc, gplus_0, innerDim_0e);
+                  Value result_1 =
+                      b.create<arith::MulFOp>(loc, gplus_1, innerDim_1e);
+                  Value lower_0 =
+                      b.create<arith::FPToSIOp>(loc, int64type, result_0);
+                  Value lower_1 =
+                      b.create<arith::FPToSIOp>(loc, int64type, result_1);
+                  Value oneInt = b.create<arith::ConstantOp>(
+                      loc, b.getIntegerAttr(int64type, 1));
+                  Value upper_0 =
+                      b.create<arith::AddIOp>(loc, int64type, lower_0, oneInt);
+                  Value upper_1 =
+                      b.create<arith::AddIOp>(loc, int64type, lower_1, oneInt);
+                  Value notValid_0 = createGreaterThan(b, loc, int64type,
```

Comment:
You need to pass in either a signed or unsigned `int64Type`, specifically the signedness of the grid sample index element type. This is so that the GT operator knows whether to make a signed or unsigned version. This is why the presubmits are failing.

Reply:
need some enlightenment here ;  in the moment I dont see what I am missing